src/Extractor.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TelemetryData;
+
+use Keboola\Component\Logger;
+use Keboola\Component\Manifest\ManifestManager;
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvWriter;
+use Keboola\Datatype\Definition\Exception\InvalidTypeException;
+use Keboola\Datatype\Definition\GenericStorage;
+use Keboola\Datatype\Definition\MySQL;
+use Keboola\TelemetryData\ValueObject\Column;
+use Keboola\TelemetryData\ValueObject\Table;
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\RetryProxy;
+use \PDOException;
+use \PDOStatement;
+use \PDO;
+use Symfony\Component\Filesystem\Filesystem;
+use Keboola\Component\Manifest\ManifestManager\Options\OutTableManifestOptions;
+
+class Extractor
+{
+
+    private DbConnector $dbConnector;
+
+    private Config $config;
+
+    private Logger $logger;
+
+    private ManifestManager $manifestManager;
+
+    private string $datadir;
+
+    public function __construct(
+        DbConnector $dbConnector,
+        Config $config,
+        Logger $logger,
+        ManifestManager $manifestManager,
+        string $datadir
+    ) {
+        $this->dbConnector = $dbConnector;
+        $this->config = $config;
+        $this->logger = $logger;
+        $this->datadir = $datadir;
+        $this->manifestManager = $manifestManager;
+    }
+
+    public function extractData(): void
```

Comment:
Bolo by dobre (iba navrh) oddelit generovanie SQL do osobitnej metody, bolo by to prehladnejsie.

Reply:
jj to bych mohl 👍  je to tam